File: In C/searching_algorithm.c
// Searching Algorithms like linear search and binary search
#include <stdio.h>


int main()
{
    int linearSearch(int [],int,int);
    int binarySearch(int [],int,int);
    int lin_array[10]={23,45,1,2,56,7,8,10};  //unsorted array used for linear search
    int bin_array[10]={10,20,30,40,50,60,70,80};  //sorted array for binary seach
    int element=100;
    int size=8;
    // int index=linearSearch(lin_array,element,size);
    int index=binarySearch(bin_array,size,element);
    if(index==-1){
        printf("Element not found");
    }
    else{
        printf("Element %d found at %d",element,index);
    }

    return 0;
}

int linearSearch(int array[],int element,int size){
    for(int i=0;i<size;i++){
        if(array[i]==element){
            return i;
        }
    }
    return -1;
}

int binarySearch(int array[],int size,int element){
    int mid,low,high;
    low=0;
    high=size-1;
    while(low<=high){
        mid=(low+high)/2;
        if(array[mid]==element){
            return mid;
        }
        if(array[mid]<element){
            low=mid+1;
        }
        else{
            high=mid-1;
        }
    }
    return -1;
}

File: In Java/src/dsa/searching_algorithm.java
package dsa;
/*
 * Implementing the searching algorithm like binary search and linear search
 */

class S_Algorithm{
    public int linearSearch(int array[],int element){
        for(int i=0;i<array.length;i++){
            if(array[i]==element){
                return i;
            }
        }
        return -1;
    }

    public int binarySearch(int array[],int element){
        int mid,low,high;
        low=0;
        high=array.length-1;
        while(low<=high){
            mid=(low+high)/2;
            if(array[mid]==element){
                return mid;
            }
            if(array[mid]>element){
                high=mid-1;
            }
            else{
                low=mid+1;
            }
        }
        return -1;
    }
}

public class searching_algorithm {
    public static void main(String[] args) {
        S_Algorithm sa=new S_Algorithm();
        @SuppressWarnings("unused")
		int lin_arr[]={12,34,112,56,78,89};
        int bin_arr[]={10,20,30,40,50,60,70};
        int element=20;
        // int index=sa.linearSearch(lin_arr, element);
        int index=sa.binarySearch(bin_arr, element);
        if(index==-1){
            System.out.println("Element Not Found");
        }
        else{
            System.out.println("Element "+element+" found at index "+index);
        }

    }
}
